LR_Tourist/TouristWebApp/Controllers/HotelController.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BLL.Model;
+using BLL.Services;
+using DA;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace TouristWebApp.Controllers
+{
+    public class HotelController : Controller
+    {
+        private readonly HotelManagementService _hotelManagementService;
+        public HotelController(HotelManagementService hotelManagementService)
+        {
+            _hotelManagementService = hotelManagementService;
+        }
+        // GET: Hotel
+        public async Task<IActionResult> Index()
+        {
+            return View(await _hotelManagementService.GetItems());
+        }
+
+        // GET: Hotel/Details/5
+        public ActionResult Details(int id)
+        {
+            return View();
+        }
+
+        // GET: Hotel/Create
+        public ActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Hotel/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(IFormCollection collection)
+        {
+            try
+            {
+                // TODO: Add insert logic here
+                var hotel = new Hotel
+                {
+                    Name = collection["Name"],
+                    Phone = collection["Phone"],
+                    Star = Convert.ToInt32(collection["Star"]),
+                };
+                await _hotelManagementService.Create(hotel);
+                return RedirectToAction(nameof(Index));
+            }
+            catch
+            {
+                return View();
```

Comment:
Any errors should be passed here for providing some human-readable info about the issue?

Reply:
fixed